### PR TITLE
improve the logging and error messages in un/assign handler

### PR DESCRIPTION
**Pull request by grodrigues3**
I noticed a few log messages that were wrong.  

Not sure if this is the best way to handle the error from `AssignPR` from the githubapi.

### CODE REVIEW

File: mungegithub/mungers/assign-unassign-handler.go [hunk 1]
```diff
@@ -84,10 +85,14 @@ func (h AssignUnassignHandler) Munge(obj *github.MungeObject) {
 
 	//get ALL (not just leaf) the people that could potentially own the file based on the blunderbuss.go implementation
 	potentialOwners, _ := getPotentialOwners(*obj.Issue.User.Login, h.features, fileList, false)
+	glog.Infof("potential owners %+v", potentialOwners)
```
Comment (apelisse) [inline mungegithub/mungers/assign-unassign-handler.go]: That's good for debug, but it's too verbose for normal run

File: mungegithub/mungers/assign-unassign-handler.go [hunk 2]
```diff
@@ -84,10 +85,14 @@ func (h AssignUnassignHandler) Munge(obj *github.MungeObject) {
...
 
 	toAssign, toUnassign := h.getAssigneesAndUnassignees(obj, comments, fileList, potentialOwners)
 	for _, username := range toAssign.List() {
-		obj.AssignPR(username)
+		err := obj.AssignPR(username)
+		if err != nil {
+			glog.Infof("Could not assign %v to PR %v because %v", username, obj.Issue.Number, err)
```
Comment (apelisse) [inline mungegithub/mungers/assign-unassign-handler.go]: That seems to be an error, maybe it should be `glog.Error`

**Review by apelisse — CHANGES_REQUESTED** [review]

File: mungegithub/mungers/assign-unassign-handler.go [hunk 3]
```diff
@@ -87,9 +88,18 @@ func (h AssignUnassignHandler) Munge(obj *github.MungeObject) {
...
+			glog.Errorf("Could not assign %v to PR %v because %v", username, obj.Issue.Number, err)
+		}
+	}
+	for _, username := range toUnassign.List() {
+		err := obj.UnassignPR(username)
+		if err != nil {
+			glog.Errorf("Could not unassign %v from PR %v because %v", username, obj.Issue.Number, err)
+		}
 	}
-	obj.UnassignPR(toUnassign.List()...)
```
Comment (apelisse) [inline mungegithub/mungers/assign-unassign-handler.go]: Why did you remove that?

Comment (grodrigues3) [inline mungegithub/mungers/assign-unassign-handler.go]: see line 97.  Instead of removing them all at once, it's done a person by person basis to 1) be consistent with the assign call and 2) so that the error is specific to a single person that wrote `/unassign` rather than for a group

Comment (apelisse) [inline mungegithub/mungers/assign-unassign-handler.go]: I'd much rather the consistency be fixed the other way around: have `AssignPR` take a list than `Unassign` take individual. I'm not too worried about the error not reporting exactly what individual couldn't be assigned (if it's not already reported by github)

Comment (grodrigues3) [inline mungegithub/mungers/assign-unassign-handler.go]: As I was re implementing it as you described, I realize another downside.  If assigns and unassigns are aggregated, then if someone not on the org/project comments `/assign` or `/unassign` and github's api returns an error, they effectively prevent everyone else from being assigned/unassigned.

File: mungegithub/mungers/assign-unassign-handler.go [hunk 4]
```diff
@@ -87,9 +88,18 @@ func (h AssignUnassignHandler) Munge(obj *github.MungeObject) {
 
 	toAssign, toUnassign := h.getAssigneesAndUnassignees(obj, comments, fileList, potentialOwners)
 	for _, username := range toAssign.List() {
-		obj.AssignPR(username)
+		err := obj.AssignPR(username)
```
Comment (eparis) [inline mungegithub/mungers/assign-unassign-handler.go]: Typically we do the glog.Errorf() in the github.go function and be silent here...

File: mungegithub/mungers/assign-unassign-handler.go [hunk 5]
```diff
@@ -87,9 +88,18 @@ func (h AssignUnassignHandler) Munge(obj *github.MungeObject) {
...
+		if err != nil {
+			glog.Errorf("Could not assign %v to PR %v because %v", username, obj.Issue.Number, err)
+		}
+	}
+	for _, username := range toUnassign.List() {
+		err := obj.UnassignPR(username)
+		if err != nil {
+			glog.Errorf("Could not unassign %v from PR %v because %v", username, obj.Issue.Number, err)
```
Comment (eparis) [inline mungegithub/mungers/assign-unassign-handler.go]: same comment here about where to do the Errorf

**Comment by grodrigues3** [comment]
Closing this PR because changing the behavior to do the assigns all at once has unintended consequences (it's possible to prevent anyone form being assigned with one non project member writing `/assign` )
